x/supplier/keeper/msg_server_submit_proof.go

```diff
@@ -4,55 +4,130 @@ import (
 	"context"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 
-	"github.com/pokt-network/poktroll/x/supplier/types"
+	suppliertypes "github.com/pokt-network/poktroll/x/supplier/types"
 )
 
-func (k msgServer) SubmitProof(goCtx context.Context, msg *types.MsgSubmitProof) (*types.MsgSubmitProofResponse, error) {
+func (k msgServer) SubmitProof(goCtx context.Context, msg *suppliertypes.MsgSubmitProof) (*suppliertypes.MsgSubmitProofResponse, error) {
 	// TODO_BLOCKER: Prevent Proof upserts after the tokenomics module has processes the respective session.
 	// TODO_BLOCKER: Validate the signature on the Proof message corresponds to the supplier before Upserting.
-
 	ctx := sdk.UnwrapSDKContext(goCtx)
+	logger := k.Logger(ctx).With("method", "SubmitProof")
+	logger.Debug("submitting proof")
+
+	/*
+		TODO_INCOMPLETE: Handling the message
+
+		## Actions (error if anything fails)
+		1. Retrieve a fully hydrated `session` from on-chain store using `msg` metadata
+		2. Retrieve a fully hydrated `claim` from on-chain store using `msg` metadata
+		3. Retrieve `relay.Req` and `relay.Res` from deserializing `proof.ClosestValueHash`
+
+		## Basic Validations (metadata only)
+		1. proof.sessionId == claim.sessionId
+		2. msg.supplier in session.suppliers
+		3. relay.Req.signer == session.appAddr
+		4. relay.Res.signer == msg.supplier
+
+		## Msg distribution validation (governance based params)
+		1. Validate Proof submission is not too early; governance-based param + pseudo-random variation
+		2. Validate Proof submission is not too late; governance-based param + pseudo-random variation
+
+		## Relay Mining validation
+		1. verify(proof.path) is the expected path; pseudo-random variation using on-chain data
+		2. verify(proof.ValueHash, expectedDiffictulty); governance based
```

Comment:
Does this step include validation of the relay `proof.ValueHash` itself? Signatures etc? Ensuring the relay is an actual relay should be here, if I have missed that ignore this.